test/jest/pre-init-utils.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { processReadyQueue } from '../../src/pre-init-utils';
+
+describe('pre-init-utils', () => {
+    describe('#processReadyQueue', () => {
+        it('should return an empty array if readyQueue is empty', () => {
+            const result = processReadyQueue([]);
+            expect(result).toEqual([]);
+        });
+
+        it('should process functions passed as arguments', () => {
+            const functionSpy = jest.fn();
+            const readyQueue: Function[] = [functionSpy, functionSpy, functionSpy];
+            const result = processReadyQueue(readyQueue);
+            expect(functionSpy).toHaveBeenCalledTimes(3);
+            expect(result).toEqual([]);
+        });
+
+        it('should process functions passed as arrays', () => {
+            const functionSpy = jest.fn();
+            (window.mParticle as any) = {
+                fakeFunction: functionSpy,
+            };
+            const readyQueue = [['fakeFunction']];
+            processReadyQueue(readyQueue);
+            expect(functionSpy).toHaveBeenCalled();
+        });
+
+        it('should process functions passed as arrays with arguments', () => {
+            const functionSpy = jest.fn();
+            (window.mParticle as any) = {
+                fakeFunction: functionSpy,
+                args: () => {},
+            };
+            const readyQueue = [['fakeFunction', 'args']];
+            processReadyQueue(readyQueue);
+            expect(functionSpy).toHaveBeenCalledWith('args');
+        });
+
+        it('should process arrays passed as arguments with multiple methods', () => {
+            const functionSpy = jest.fn();
+            (window.mParticle as any) = {
+                fakeFunction: {
+                    anotherFakeFunction: functionSpy,
+                },
+            };
+            const readyQueue = [['fakeFunction.anotherFakeFunction', 'foo']];
+            processReadyQueue(readyQueue);
+            expect(functionSpy).toHaveBeenCalledWith('foo');
+        });
```

Comment:
is this multiple methods?  it seems like it's a single method, but 1 layer deeper.
Should the test instead be:

```suggestion
        it('should process arrays passed as arguments with multiple methods', () => {
            const functionSpy = jest.fn();
            const functionSpy2 = jest.fn();
            (window.mParticle as any) = {
                fakeFunction: functionSpy,
                anotherFakeFunction: functionSpy2,
            };
            const readyQueue = [['fakeFunction', 'foo'], [anotherFakeFunction', 'foo2']];
            processReadyQueue(readyQueue);
            expect(functionSpy).toHaveBeenCalledWith('foo');
            expect(functionSpy).toHaveBeenCalledWith('foo');
        });
```

(This may not run as I just did the suggestion inline here, but I think you get the point)

Reply:
or actually I think I see what you are doing here. It's just that the test name is incorrect.

I think the suggestion I gave above can be its own test though so actually test multiple methods inside of the process queue when they are in array form.